docs/docs/04-standard-library/01-cloud/endpoint.md

```diff
@@ -0,0 +1,72 @@
+---
+title: Endpoint
+id: endpoint
+description: A resource that represents an endpoint.
+keywords:
+  [
+    Wing reference,
+    Wing language,
+    language,
+    Wing standard library,
+    Wing programming language,
+    Endpoint,
+    URL,
+  ]
+sidebar_position: 1
+---
+
+The `cloud.Endpoint` resource holds a URL and outputs it.
+
+## Usage
+
+```ts playground
+bring cloud;
+
+let endpoint = new cloud.Endpoint("https://example.com");
+```
```

Comment:
I'm not sure I understand the reasoning for renaming this resource from `cloud.Output` to `cloud.Endpoint`. Can you elaborate @eladcon @eladb?

As a user, when I see `new cloud.Endpoint("http://example.com")`, as a user it looks as though the app is going to perform some work to expose a resource (file, api, socket, etc.) on the networked address, just like how `cloud.Api` does. But that doesn't seem to be how the resource works.

The intent of the original `cloud.Output` API was to _expose simple data as the **output** of a Wing program_. I think this is a more natural mental model. Developers are familiar with the concepts of inputs and outputs. The name "output" will also make the resource easier to pick up for users coming from other tools (see CloudFormation Outputs, Pulumi outputs, Terraform outputs, etc.). I'm not sure if there's enough differentiated functionality here to justify deviating from the existing norms.

Reply:
An endpoint is more than just an output. Our use case is to be able to show endpoints in the console, expose them from preview environments and from local dev machines, maybe bind them to domains, etc.

So they need to be explicitly modeled.

I am not against also introducing `cloud.Output` (now or later) and define an output for all endpoints, but it's not required for the endpoint use case.

I hope that makes sense.

Reply:
Created https://github.com/winglang/wing/issues/5096 to cover endpoints and renamed back the original issue to Output

Reply:
Ok - thanks for explaining. Yes I think separating them (and/or having each Endpoint create an Output) sounds ok to me.

I'm curious to see how these resources that only have a semantic use in the context of Wing Cloud or Wing Console  end up evolving... (the resources in `bring ui` are another example of this)

Reply:
Geeking out...

They have a semantic context, period. Wing Cloud uses these semantics to implement an experience, but any tool can leverage these semantics in the future of course.